torch/futures/__init__.py

```diff
@@ -15,7 +33,7 @@ def wait(self):
         """
         return super().wait()
 
-    def then(self, callback):
+    def then(self, callback : Callable[[T], S]) -> torch._C.Future:
```

Comment:
The first argument type should be `Future[T]` instead of `T`?

Reply:
Do we need new tests to verify types?

Reply:
Umm, so you seem to be right as far as the code is concerned... but why does callback take a future? That doesn't make any sense!

Reply:
@mrshenli Added test/test_futures.py to the list of type-checked code, so those types of typos are now detected. Also, added https://github.com/pytorch/pytorch/pull/41826 that actually enables running test_futures.py as part of CI system.
But receiving future in the callback looks a bit unusual.

Reply:
(unresolved the convo to make sure @mrshenli can see it)